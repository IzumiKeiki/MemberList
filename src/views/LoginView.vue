<script setup lang="ts">
import { useAuthStore } from '../stores/authStore'
import { useRouter } from 'vue-router'
import type { RouteLocationNormalized } from 'vue-router'

const authStore = useAuthStore()
const router = useRouter()

const submit = () => {
  authStore.login()
  if (authStore.confirm) {
    router.replace('/home')
  }
}

const beforeRouteEnter = (
  to: RouteLocationNormalized,
  from: RouteLocationNormalized,
  next: () => void
) => {
  if (to.path === '/home' && !authStore.confirm) {
    router.push('/')
  } else {
    next()
  }
}

router.beforeEach(beforeRouteEnter)
</script>

<template>
  <div class="login">
    <h1>Login</h1>
    <p v-if="authStore.showError">
      ID or Password was wrong! <br />
      You can try "abc" for ID and "pass" for Password
    </p>
    <form @submit.prevent="submit">
      <label for="id">ID :</label>
      <input id="id" type="text" v-model="authStore.id" />
      <label for="pass">Password :</label>
      <input id="pass" type="password" v-model="authStore.pass" />
      <button type="submit">Confirm</button>
    </form>
  </div>
</template>

<style scoped>
div.login {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

h1 {
  padding: 1rem 0;
}
.login p {
  color: red;
  padding: 1rem 0;
}

form {
  height: 250px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: space-between;
}
</style>
