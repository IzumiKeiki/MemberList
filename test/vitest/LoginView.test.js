import { mount } from '@vue/test-utils'
import { beforeEach, describe, it, expect } from 'vitest'
import { setActivePinia, createPinia } from 'pinia'

import LoginView from '../../src/views/LoginView.vue'
import router from '../../src/router'

describe('LoginView', () => {
  beforeEach(() => {
    setActivePinia(createPinia())
  })

  it('renders correctly', () => {
    expect(LoginView).toBeTruthy()
    const wrapper = mount(LoginView, {
      global: {
        plugins: [router]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has a title and a button', () => {
    const wrapper = mount(LoginView, {
      global: {
        plugins: [router]
      }
    })
    expect(wrapper.text()).toContain('Login')
    expect(wrapper.find('button').exists()).toBe(true)
  })

  it('should show error message when the input is invalid', async () => {
    const wrapper = mount(LoginView, {
      global: {
        plugins: [router]
      }
    })
    console.log('ERROR Before button click:', wrapper.vm.authStore.showError)

    const idInput = wrapper.find('#id')
    const passInput = wrapper.find('#pass')
    idInput.setValue('id')
    passInput.setValue('pass')
    wrapper.find('form').trigger('submit.prevent')
    await wrapper.vm.$nextTick()
    console.log('ERROR After button click:', wrapper.vm.authStore.showError)

    expect(wrapper.find('p').exists()).toBe(true)
  })
})
