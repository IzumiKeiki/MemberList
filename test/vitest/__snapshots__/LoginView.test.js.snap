// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginView > renders correctly 1`] = `
"<div data-v-45f5edd7=\\"\\" class=\\"login\\">
  <h1 data-v-45f5edd7=\\"\\">Login</h1>
  <!--v-if-->
  <form data-v-45f5edd7=\\"\\"><label data-v-45f5edd7=\\"\\" for=\\"user\\">ID :</label><input data-v-45f5edd7=\\"\\" id=\\"user\\" type=\\"text\\"><label data-v-45f5edd7=\\"\\" for=\\"pass\\">Password :</label><input data-v-45f5edd7=\\"\\" id=\\"pass\\" type=\\"password\\"><button data-v-45f5edd7=\\"\\" type=\\"submit\\">Confirm</button></form>
</div>"
`;
